nixos/modules/services/networking/prosody.nix

```diff
@@ -475,15 +487,15 @@ in
         '') cfg.virtualHosts) }
     '';
 
-    users.extraUsers.prosody = {
+    users.extraUsers."${cfg.user}" = {
```

Comment:
I think the user should only be created if `cfg.user == prosody`. Same for the group.